hal/watchdog_api.h

```diff
@@ -0,0 +1,116 @@
+/** \addtogroup hal */
+/** @{*/
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_WATCHDOG_API_H
+#define MBED_WATCHDOG_API_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+typedef struct
+{
+  // Refresh value for the watchdog in milliseconds. The maximum value of this
+  // setting is platform dependent, to find the maximum value for the current
+  // platform call hal_watchdog_get_max_timeout(void)
+  uint32_t timeout_ms;
+  // Enables the window watchdog. This will cause the API to run the window
```

Comment:
windowed watchdog ? 

It may be a good thing to detail modes of operations supported by the API.